lua/plenary/window/border.lua

```diff
@@ -147,8 +147,11 @@ function Border:change_title(new_title)
   vim.api.nvim_buf_set_lines(self.bufnr, 0, -1, false, self.contents)
 end
 
-function Border:new(content_bufnr, content_win_id, content_win_options, border_win_options)
-  assert(type(content_win_id) == "number", "Must supply a valid win_id. It's possible you forgot to call with ':'")
+-- Sets the size and position of the given Border.
+-- Can be used to create a new window (with `create_window = true`)
+-- or change an existing one
+function Border:set_size(content_win_options, border_win_options, create_window)
+  create_window = (create_window ~= nil) and create_window or false
```

Comment:
In general, `vim.F.if_nil(value, default)` I guess would be more readable. But what about

```lua
  if self.win_id then
    vim.api.nvim_win_set_config(self.win_id, nvim_win_config)
  else
    self.win_id = vim.api.nvim_open_win(self.bufnr, false, nvim_win_config)
  end
```

which should allow you to get rid of the `create_window` argument entirely.